spec/inheritance_spec.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+RSpec.describe "inheritance" do # rubocop:disable RSpec/DescribeClass
+  subject do
+    Class.new(inherited_klass)
+  end
+
+  context "with simple inheritance" do
+    let(:inherited_klass) do
+      Class.new do
+        def self.class_method
+          "class_method"
+        end
+
+        def instance_method
+          "instance_method"
+        end
+      end
+    end
+
+    it { expect(subject.class_method).to eq("class_method") }
```

Comment:
These two tests were unrelated to the specific issue, just seemed helpful to include